src/realm/obj.cpp

```diff
@@ -1168,6 +1173,35 @@ REALM_FORCEINLINE void Obj::sync(Node& arr)
     }
 }
 
+// helper functions for filtering out calls to set_string_interner()
+template <class T>
+inline void Obj::set_string_interner(T&, ColKey)
```

Comment:
I think `values.set_string_interner(m_table->get_string_interner(col_key));` can just go in the main template method. The interface is basically the same, isn't it?

Reply:
It is an optimization to only call that method for strings and mixed.